File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.kt
class Solution {
fun answerQueries(nums: IntArray, queries: IntArray): IntArray {
    nums.sort()
    for (i in 1..nums.lastIndex) nums[i] += nums[i-1]
    return queries.map { 
        val search = nums.binarySearch(it)
        if (search < 0)-search-1 else search+1
    }.toIntArray()
}
}

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @return {number[]}
 */
var answerQueries = function(nums, queries) {
    const qLen = queries.length
    const nLen = nums.length
    nums = nums.sort((a, b) => a - b)
    let answer = []
    for(let i = 0; i < qLen; i++) {
        let target = queries[i]
        let len = 0
        let max = 0
        let temp = []
        for(let j = 0; j < nLen; j++) {
            if(target >= nums[j]) {
                target -= nums[j]
                len++
            }   
            max = Math.max(max, len)
        }
        answer.push(max)    
    }
    return answer === [] ? [0] : answer
};